subsys/bluetooth/host/conn.c

```diff
@@ -79,6 +80,7 @@ NET_BUF_POOL_FIXED_DEFINE(frag_pool, CONFIG_BT_L2CAP_TX_FRAG_COUNT,
 
 #if defined(CONFIG_BT_SMP) || defined(CONFIG_BT_BREDR)
 const struct bt_conn_auth_cb *bt_auth;
+sys_slist_t bt_auth_info_cbs = SYS_SLIST_STATIC_INIT(&bt_auth_info_cbs);
```

Comment:
Missing `static` in the beginning of the line? 

Btw, did you consider using a linker section? That would reduce the size by a pointer and make is possible to place these structs in ROM instead of RAM. Only downside is that we loose the possibility for the `unregister()` API, but I'm failing to think of good use cases for it right now.

Reply:
The reason why `static` was omitted is that the callbacks are actually never called from this file, but rather from `hci_core.c`, `ssp.c` and `smp.c` and  I decided to store it here, as the `struct bt_conn_auth_cb *bt_auth;` is stored here. 

Regarding a linker section instead, I have not yet considered that. I am absolutely fine with using that instead. 
The only use case that I have where you'd want to unregister these calls, is if you have a dynamically registered GATT service (we have some of those in LE Audio), where you may want to unregister the callbacks as you unregister the service, but that can also be done internally in the service implementation using a simple boolean to ignore callbacks.